fastchat/serve/model_worker.py

```diff
@@ -246,10 +246,21 @@ def get_embeddings(self, params):
             input_ids = tokenizer.encode(params["input"], return_tensors="pt").to(
                 self.device
             )
-            model_output = self.model(input_ids, output_hidden_states=True)
-            is_chatglm = "chatglm" in str(type(self.model)).lower()
+            model_name = str(type(self.model)).lower()
+            is_chatglm = "chatglm" in model_name
+            is_t5 = "t5" in model_name
+            if is_t5:
+                model_output = self.model(
+                    input_ids, decoder_input_ids=input_ids, output_hidden_states=True
+                )
+            else:
+                model_output = self.model(input_ids, output_hidden_states=True)
             if is_chatglm:
                 data = (model_output.hidden_states[-1].transpose(0, 1))[0]
+            elif is_t5:
+                # model uses Seq2SeqLMOutput from transformers to encode/decode
+                data = model_output.decoder_hidden_states[-1][0]
+                # data = (model_output.decoder_hidden_states[-1].transpose(0, 1))[0]
```

Comment:
@merrymercy do you know what the significance between `model_output.decoder_hidden_states[-1][0]` and `(model_output.decoder_hidden_states[-1].transpose(0, 1))[0]` ? They both return a vector, not sure which is desired here as I'm new to this project.

Reply:
@cterrazas2 `model_output.decoder_hidden_states[-1]` is the last hidden state of the model, and most outputs follow [batch, seq, hidden_size]. See [model output](https://huggingface.co/docs/transformers/main_classes/output#transformers.modeling_outputs.CausalLMOutput.hidden_states)
However, chatglm doesn't follow the rule. It is like [batch, hidden_size, seq]. Therefore, we have to transpose it.